src/gui/attributetable/qgsdualview.cpp

```diff
@@ -979,6 +981,46 @@ void QgsDualView::showViewHeaderMenu( QPoint point )
   connect( sort, &QAction::triggered, this, [ = ]() {modifySort();} );
   mHorizontalHeaderMenu->addAction( sort );
 
+  mConfig.update( mLayer->fields() );
+  // get layer field index from column name
+  const int fieldIndex = mLayer->fields().indexFromName( mConfig.columns().at( col ).name );
+  const int fieldOrigin  = mLayer->fields().fieldOrigin( fieldIndex );
+
+  bool fieldCalculatorEnabled = false;
+
+  mHorizontalHeaderMenu->addSeparator();
+  if ( fieldOrigin == 4 )
+  {
+    QAction *updateField = new QAction( tr( "Update &Field..." ), mHorizontalHeaderMenu );
+    connect( updateField, &QAction::triggered, this, &QgsDualView::updateField );
+    updateField->setData( col );
+    mHorizontalHeaderMenu->addAction( updateField );
+
+    fieldCalculatorEnabled = true;
+    updateField->setEnabled( fieldCalculatorEnabled );
+  }
+  else
+  {
+    if ( fieldOrigin == 2 )
+    {
+      int srcFieldIndex;
+      const QgsVectorLayerJoinInfo *info = mLayer->joinBuffer()->joinForFieldIndex( fieldIndex, mLayer->fields(), srcFieldIndex );
+
+      if ( info && info->isEditable() )
+        fieldCalculatorEnabled = true;
+    }
+
+    if ( fieldOrigin == 1 || fieldOrigin == 3 )
+      fieldCalculatorEnabled = true;
+
+    QAction *fieldCalculator = new QAction( tr( "Open &Field Calculator..." ), mHorizontalHeaderMenu );
```

Comment:
```suggestion
    QAction *fieldCalculator = new QAction( tr( "Open &Field Calculator…" ), mHorizontalHeaderMenu );
```
(use the real horizontal ellipsis unicode character, not 3 dots)